dlrover/python/master/node/training_node.py

```diff
@@ -184,6 +184,26 @@ def __init__(
         self._lock = threading.Lock()
         self._node_id_iter = itertools.count(len(self._nodes))
         self._node_rank_iter = itertools.count(len(self._nodes))
+        self._pending_nodes: List[Node] = []
+
+    @property
+    def pending_nodes(self):
+        pending_nodes = [node.name for node in self._pending_nodes]
+        return pending_nodes
+
+    @property
+    def first_pending_node(self):
+        if len(self._pending_nodes) == 0:
+            return ""
+        first_pending_node = min(
+            self._pending_nodes, key=lambda x: x.create_time  # type: ignore
+        )
+        return first_pending_node.name
+
+    @property
+    def cur_nodes(self):
+        cur_nodes = [node.name for node in self._nodes.values()]
```

Comment:
```return [node.name for node in self._nodes.values()]```

Reply:
We could print the return value directly in this way.